sway-core/src/type_system/substitute/subst_map.rs

```diff
@@ -148,6 +153,7 @@ impl TypeSubstMap {
     /// methods, etc, that are implemented for the type of `superset` so that
     /// they can be used for `subset`.
     pub(crate) fn from_superset_and_subset(
+        engines: &Engines,
         type_engine: &TypeEngine,
         decl_engine: &DeclEngine,
```

Comment:
As mentioned above, adding `engines` as a parameter means we can remove `type_engine` and `decl_engine` and extract them directly from `engines` in the function body instead.

Reply:
Done.